### PR TITLE
Add timeusage django app to produce timespent plots

### PR DESCRIPTION
**Pull request by rigperro**
# Description

# Testing

- [ ] I created new unit tests
- [ ] All unit tests pass
- [x] I have tested manually

# Have you updated the README or other relevant documentation?

- [ ] Yes
- [?] Not yet. I will do it next.
- [ ] Not relevant

### CODE REVIEW

File: jutut/settings.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
     'bootstrapform',
     'bootstrapform_jinja',
     'django_filters',
+    'plotly',
```
Comment (raphendyr) [inline jutut/settings.py:55]: I think this is not needed. I have not checked, but plotly shouldn't have django app.

Comment (raphendyr) [inline jutut/settings.py:55]: I think this is not needed. I have not checked, but plotly shouldn't have django app.

File: jutut/urls.py [hunk 2]
```diff
@@ -4,10 +4,12 @@
 import core.urls
 import accounts.urls
 import feedback.urls
+import timeusage.urls
 
 
 urlpatterns = [
     url(r'^', include(feedback.urls)),
+    url(r'^', include(timeusage.urls)),
```
Comment (raphendyr) [inline jutut/urls.py:12]: This could be added with the prefix, i.e., `url(r'^timeusage/', include(timeusage.urls)),`.

File: requirements.txt [hunk 3]
```diff
@@ -15,6 +15,7 @@ psycopg2-binary>=2.7.4
 python-memcached==1.58
 raphendyr-django-essentials==1.3.0
 ansi2html==1.2.0
+plotly==4.5.0
```
Comment (raphendyr) [inline requirements.txt:18]: Should use relative requirement `plotly ~= 4.5.0`, so point releases are used instead.

File: timeusage/templates/time_usage.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	<script src="{% static "filterbar.js" %}"></script>{# extra js for feedback filter tool #}
+{% endblock %}
+
+{% block content %}
+	<h1>Time usage for {{course}}</h1>
+
+	{% include "_filter_bar.html" %}
+
+	{% autoescape off %}
+		<p>{{ c_plot }}<p>
```
Comment (raphendyr) [inline timeusage/templates/time_usage.html:17]: These should be pullet via ajax request, but could be merged as is for now.

File: timeusage/urls.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from django.conf.urls import url
+from . import views
+from .apps import TimeUsageConfig
+
+app_name = TimeUsageConfig.name
+urlpatterns = [
+    url(r'^timeusage/(?P<course_id>\d+)/$',
```
Comment (raphendyr) [inline timeusage/urls.py:7]: This app should be mounted with timeusage in the main url config. See the other comment.

File: timeusage/views.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    model = Feedback
+    template_name = "time_usage.html"
+
+    def plot_times(self):
+        feedbacks = self.get_queryset()
+        feedbacks_by_exercise = {}
+        for f in feedbacks:
+            feedbacks_by_exercise.setdefault(f.exercise, []).append(f)
+        
+        x_data = []
```
Comment (raphendyr) [inline timeusage/views.py:21]: I'm pretty sure this can be written prettier, but that is _not_ important of course.

File: feedback/filters.py [hunk 7]
```diff
@@ -193,7 +193,7 @@ class FeedbackFilter(django_filters.FilterSet):
                                           widget=forms.CheckboxSelectMultiple())
     flags = FlagFilter(label=_("Flags"))
     tags = ColortagIEAndOrFilter(queryset=FeedbackTag.objects.none(), label=_("Tags"))
-    student_tags = ColortagIEAndOrFilter(queryset=StudentTag.objects.none(), label=_("Student tags"))
+    student_tags = ColortagIEAndOrFilter(queryset=StudentTag.objects.none(), field_name='student__tags', label=_("Student tags"))
```
Comment (raphendyr) [inline feedback/filters.py:196]: This is already in master? https://github.com/Aalto-LeTech/mooc-jutut/commit/678e70268f77026feb26e05b935810f9a3f0b263

File: timeusage/views.py [hunk 8]
```diff
@@ -6,18 +6,17 @@
 from plotly.offline import plot
 from plotly.graph_objs import Bar, Figure, Layout
 
-
```
Comment (raphendyr) [inline timeusage/views.py]: 2 spaces is good

File: timeusage/views.py [hunk 9]
```diff
@@ -6,18 +6,17 @@
...
 class TimeUsageView(ManageCourseMixin, ListView):
 
     model = Feedback
     template_name = "time_usage.html"
-
```
Comment (raphendyr) [inline timeusage/views.py]: this space was good

File: timeusage/views.py [hunk 10]
```diff
@@ -6,18 +6,17 @@
 from plotly.offline import plot
 from plotly.graph_objs import Bar, Figure, Layout
 
-
 class TimeUsageView(ManageCourseMixin, ListView):
```
Comment (raphendyr) [inline timeusage/views.py:10]: this space is not needed, but there could be documentation string here though.

File: timeusage/views.py [hunk 11]
```diff
@@ -44,8 +43,10 @@ def plot_times(self):
 
         chapter_times = plot(
             Figure(
-                [Bar(name = '75th%', x = x_data, y = perc75_data, marker_color = '#00adf0'),
-                Bar(name = 'median', x = x_data, y = md_data, marker_color = '#0077b4')],
+                [
+                    Bar(name = '75th%', x = x_data, y = perc75_data, marker_color = '#00adf0'),
```
Comment (raphendyr) [inline timeusage/views.py:47]: class/function arguments don't typically have space around `=`

File: timeusage/templates/time_usage.html [hunk 12]
```diff
@@ -0,0 +1,21 @@
+{% extends "manage/base.html" %}
+{% load static %}
+{% load dictiterators %}
+{% load i18n %}
+
+{% block head %}
+{{ block.super }}
+	<script src="{% static "filterbar.js" %}"></script>{# extra js for feedback filter tool #}
```
Comment (Mankro) [inline timeusage/templates/time_usage.html:8]: Where is `filterbar.js` since I don't see it in this pull request?

**Review by Mankro — APPROVED** [review]
Based on a quick code review, this is OK. Let's install this on a test server first.

File: timeusage/templates/time_usage.html (hunk 12 above)
Comment (Mankro) [reply timeusage/templates/time_usage.html:8]: Actually, `filterbar.js` is a part of the existing mooc-jutut code: 
https://github.com/Aalto-LeTech/mooc-jutut/blob/911ac911c3b8c8e93d9f1ce511f38a1a7c61015b/feedback/static/filterbar.js

**Pull request merged**